### DISCUSSION TITLE
Feature suggestion

### DISCUSSION THREAD

**Discussion by Kriss1670** [teslamate-org/teslamate#1452] [Ideas]
It would be great that there is a way to save the database and retrieve it easily. When user has a NAS, it would be very interesting to make this save on the NAS.
Thanks.

**Comment by DrMichael** [comment]
You are aware of https://docs.teslamate.org/docs/maintenance/backup_restore ?

(Could you make the subject perhaps more concrete please?)

**Reply by Kriss1670** [reply]
Yes, I am aware. As well as I am aware that lot of people are frightened to use command lines on a console... (as me for example 😉)
As data is the application heart, a simple way of backing and restoring core data would be way better than using console.
Something like a target place to store backups in the Settings page and a restore button aside.

**Comment by DrMichael** [comment]
@adriankumpf Wouldn't it be (easily?:-) possible to have a backup download button in Teslamate and one to upload and restore?

(I don't speak Elixir, unfortunately...)

**Reply by Kriss1670** [reply]
That was my request...
Button or link to a directory with an automatic backup every day ?

**Comment by DrMichael** [comment]
I don't see a link to a directory. That would depend completely on the environment. Also backup every day would be on the backend side, not uniquely accessible from the frontend (browser) system.

**Comment by swotai** [comment]
I think if you expose the Postgres ports, you can also do a manual dump of the tables from there.
